files/external_node_v2.rb

```diff
@@ -163,6 +165,8 @@ def generate_fact_request(certname, filename)
     rescue => e
       raise "Could not generate facts for Foreman: #{e}"
     end
+  else
+    raise 'Last facts are older than last upload time, rerun puppet agent on host to get newer facts'
```

Comment:
Raising an exception here might cause the watch process to exit entirely, as I don't think anything catches it. Shouldn't this be logged like other instances?